data-access-layer/eventmesh/LockManager.js

```diff
@@ -214,8 +214,15 @@ class LockManager {
     });
   }
 
-  _getLockType(operation) {
-    if (_.includes(CONST.APISERVER.WRITE_OPERATIONS, operation)) {
+  _getLockType(requestedOperation, plan) {
+    const supportedOperations = _.get(plan, 'async_ops_supporting_parallel_sync_ops');
```

Comment:
Rather i will check for `supported_operations` for the plan and if the `supported_operation` includes `bind_during_update` then in the method `middleware.checkBlockingOperationInProgress`, if i see if its write locked and the operation is update, then if the plan supports bind_during_update, and if the requested operation is bind, i allow it through.

Reply:
Also service instance update job fires update again as part of retry. So at any point in time two updates will happen from the job, so having the read lock would lead to pushing two updates downstream.

Reply:
As discussed 'async_ops_supporting_parallel_sync_ops' parameter in plan having the info.

Reply:
Ok.